src/components/MultiHopTrade/components/FiatRamps/RampQuotes.tsx

```diff
@@ -50,33 +49,38 @@ export const RampQuotes: React.FC<RampQuotesProps> = ({ isLoading = false, onBac
     return direction === FiatRampAction.Buy ? sellFiatAmount : sellAmount
   }, [direction, sellAmount, sellFiatAmount])
 
-  const quotesQueries = useGetRampQuotes({
+  const { queries: quotesQueries, sortedQuotes } = useGetRampQuotes({
     fiatCurrency: direction === FiatRampAction.Buy ? sellFiatCurrency : buyFiatCurrency,
     assetId: direction === FiatRampAction.Buy ? buyAsset.assetId : sellAsset.assetId,
     amount: quoteAmount,
     direction,
   })
 
   const displayQuotes = useMemo(() => {
-    const quotes = quotesQueries.map(query => query.data).filter(isSome)
-
-    if (!quotes || quotes.length === 0) return []
-
-    return quotes
-  }, [quotesQueries])
-
-  // Auto-select the first quote when quotes are available and no quote is selected
-  useEffect(() => {
-    if (displayQuotes.length > 0 && !selectedQuote) {
-      dispatch(tradeRampInput.actions.setSelectedFiatRampQuote(displayQuotes[0]))
-    }
-  }, [displayQuotes, selectedQuote, dispatch])
+    return sortedQuotes
+  }, [sortedQuotes])
 
   const isQueryLoading = useMemo(() => {
     return quotesQueries.some(query => query.isLoading) || isLoading
   }, [quotesQueries, isLoading])
 
-  if (isQueryLoading) {
+  // Auto-select the best quote when quotes are available and no quote is selected
+  // This only happens on first load or when amount changes (not on refetch)
+  useEffect(() => {
+    if (isQueryLoading && !selectedQuote) return
+
+    if (displayQuotes.length > 0) {
+      const bestQuote =
```

Comment:
Array.prototype.find() will be optimistically happy at type-level here but `bestQuote` may be undefined, probably want to sanity guard it.